KyberElasticAdapter/contracts/KyberElasticAdapter.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./UniswapV3likeAdapter.sol";
+
+interface IKyberPool {
+    function token0() external view returns (address);
+
+    function token1() external view returns (address);
+
+    function swap(
+        address recipient,
+        int256 swapQty,
+        bool isToken0,
+        uint160 limitSqrtP,
+        bytes calldata data
+    ) external returns (int256 qty0, int256 qty1);
+}
+
+contract KyberElasticAdapter is UniswapV3likeAdapter {
+    using SafeERC20 for IERC20;
+
+    mapping(address => mapping(address => address)) public tknsToPoolWL;
+
+    constructor(
+        string memory _name,
+        uint256 _swapGasEstimate,
+        uint256 _quoterGasLimit,
+        address _quoter,
+        address[] memory _whitelistedPools
+    ) UniswapV3likeAdapter(_name, _swapGasEstimate, _quoter, _quoterGasLimit) {
+        addPoolsToWL(_whitelistedPools);
+    }
+
+    function addPoolsToWL(address[] memory pools) public onlyMaintainer {
+        for (uint256 i; i < pools.length; ++i) addPoolToWL(pools[i]);
+    }
+
+    function rmPoolsFromWL(address[] memory pools) external onlyMaintainer {
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html